### ISSUE TITLE
Synchronising Reasoner should not trigger explanations to be recomputed

### ISSUE THREAD

**Issue by matentzn** [protegeproject/protege#879]
In one of the latest versions of Protege, this behaviour slipped in:

After the normal debug workflow:
1. Run reasoner
2. Compute explanation for unsatisfiablity
3. Fix reason for unsatisfiablity
4. Synchronise reasoner

Protege tries to recompute explanations (i.e. the Compute explanations dialogue pops up). This dialogue never seems to go away; I think explanations should not be recomputed automatically (if important: Reasoner ELK 0.5). It hampers the normal edit flow quite significantly; any chance we can get this fixed? Thanks a lot!

**Comment by matthewhorridge** [comment]
@matentzn are you able to provide a specific example?  I can't reproduced this with the following (simple) example:

1. Create  ```SubClassOf(:A :B)```  ```SubClassOf(:A :C)``` ```DisjointClasses(:B :C)```
2. Run reasoner and observe that ```A``` is unsatisfiable.
3. Remove ```DisjointClasses(:B :C)```
4. Synchronize the reasoner

I'm just wondering if there are many explanations in your cases, and if so whether you terminate the computations of them.

**Comment by matentzn** [comment]
maybe use a really large ontology for the test where explanation computation takes more than 5-10 sec? HPO, UBERON; if that does not work I will record a video for you.

**Comment by matthewhorridge** [comment]
@matentzn I still can't reproduce this using HP, for example.  If you could record a video that would be great.

**Comment by matentzn** [comment]
Hey Matt,

here is a [video](https://www.dropbox.com/s/8bdndg0o1663vg4/recomputing_justs.mov?dl=0). 

I computed a justification for the only unsatisfiable class that was there (actually, I queried DL query tab for owl:Nothing, then (?)).

As you can see in the video, the justification is highly complex, and it took about 4 min to compute it.

After it was computed, 

1) I close the explanation window (00:03)
2) Remove the disjoint axiom involved in the unsatisfiablity (00:17)
3) Then save the ontology, (00:21)
4) synch reasoner (cmd/ctrl +R). (00:38)

As you can see, the explanation dialogue is triggered again.
[Here](https://www.dropbox.com/s/hzrkdwhbi4emkj6/mp.owl.zip?dl=0) is the ontology I used for the above.

**Comment by matthewhorridge** [comment]
@matentzn thanks for the video and the ontology.  Really cool justifications.

I haven't managed to reproduce this.  Just wondering, how many explanations did you compute (there's more than one)?  Did you interrupt it?

**Comment by matentzn** [comment]
Yeah, I computed 1, then interrupted. Weird you don’t see the same behaviour. You also on 5.5 release version?

**Comment by matthewhorridge** [comment]
@matentzn, yes 5.5.0 release.  Will try with 1
